### PR TITLE
APIs to create cloudbackup and cloudbackup schedule for a volumegroup.

### PR DESCRIPTION
**Pull request by talakad**
CloudBackupGroupCreate API creates group snapshot of a volume group( specified
either through a group id or set of labels), and uploads these snapshots
to a cloud.

CloudBackupGroupSchedCreate API creates schedule for creating and uploading
group snapshot of a volume group.

Signed-off-by: veda <veda@portworx.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** (optional)
Closes #

**Special notes for your reviewer**:

### CODE REVIEW

File: api/client/volume/client.go [hunk 1]
```diff
@@ -632,6 +644,24 @@ func (v *volumeClient) CloudBackupSchedCreate(
 	return createResponse, nil
 }
 
+// CloudBackupGroupSchedCreate for a volume group creates a schedule to backup
+// volume group to the cloud
+func (v *volumeClient) CloudBackupGroupSchedCreate(
+	input *api.CloudBackupGroupSchedCreateRequest,
+) (*api.CloudBackupSchedCreateResponse, error) {
+	createResponse := &api.CloudBackupSchedCreateResponse{}
+	req := v.c.Post().Resource(api.OsdBackupPath + "/schedgroup").Body(input)
```
Comment (disrani-px) [inline api/client/volume/client.go]: Define the paths as a constant in api.go and use them here and api/server/volume.go

File: api/api.go [hunk 2]
```diff
@@ -420,12 +431,32 @@ type CloudBackupScheduleInfo struct {
...
+	GroupID string
+	// Labels indicates a volume group for this cloudsnap schedule
+	Labels map[string]string
 }
 
 type CloudBackupSchedCreateRequest struct {
 	CloudBackupScheduleInfo
 }
 
+type CloudBackupGroupSchedCreateRequest struct {
```
Comment (disrani-px) [inline api/api.go]: The schedules will be deleted with the same API right? Maybe use CloudBackupSchedCreateRequest (and path) for creating both too?

File: api/api.go [hunk 3]
```diff
@@ -420,12 +431,32 @@ type CloudBackupScheduleInfo struct {
...
+	// GroupID indicates the group of volumes for which cloudbackup schedule is
+	// being created
+	GroupID string
+	// Labels indicates a volume group for which this group cloudsnap schedule is
+	// being created. If this is provided GroupId is not needed and vice-versa.
+	Labels map[string]string
```
Comment (disrani-px) [inline api/api.go]: We should document here what happens when both are provided.

File: api/api.go [hunk 4]
```diff
@@ -420,12 +431,32 @@ type CloudBackupScheduleInfo struct {
...
+	// CredentialUUID is cloud credential to be used with this schedule
+	CredentialUUID string
+	// Schedule is the frequence of backup
```
Comment (disrani-px) [inline api/api.go]: typo: frequency

File: api/api.go [hunk 5]
```diff
@@ -247,6 +247,17 @@ type CloudBackupCreateRequest struct {
 	Full bool
 }
 
+type CloudBackupGroupCreateRequest struct {
+	// GroupID indicates backup request for a volumegroup with this group id
```
Comment (lpabon) [inline api/api.go]: This is either GroupID or Labels. What happens if they are both set?

File: api/api.go [hunk 6]
```diff
@@ -247,6 +247,17 @@ type CloudBackupCreateRequest struct {
 	Full bool
 }
 
+type CloudBackupGroupCreateRequest struct {
```
Comment (lpabon) [inline api/api.go]: Move all new structs to api.proto

File: volume/volume.go [hunk 7]
```diff
@@ -136,6 +136,8 @@ type QuiesceDriver interface {
 type CloudBackupDriver interface {
 	// CloudBackupCreate uploads snapshot of a volume to the cloud
 	CloudBackupCreate(input *api.CloudBackupCreateRequest) error
+	// CloudBackupGroupCreate creates and then uploads volumegroup snapshots
+	CloudBackupGroupCreate(input *api.CloudBackupGroupCreateRequest) error
```
Comment (lpabon) [inline volume/volume.go]: How is status determined? When the user calls CloudBackupCreate( src_volume_id ) they can then call CloudBackupStatus( src_volume_id )

Is there something similar for this new API?

File: api/api.go (hunk 3 above)
Comment (lpabon) [reply api/api.go]: When this is moved to api.proto use `oneof` instead, and it will be clear.

File: api/api.go [hunk 8]
```diff
@@ -247,6 +247,17 @@ type CloudBackupCreateRequest struct {
...
+	// GroupID indicates backup request for a volumegroup with this group id
+	GroupID string
+	// Labels indicates backup request for a volume group with these labels
+	Labels map[string]string
```
Comment (lpabon) [inline api/api.go]: Use `oneof` to let the caller know only GroupID or Labels are used.

File: api/api.go (hunk 6 above)
Comment (talakad) [reply api/api.go]: Create a new ticket to track this effort separately #516

File: api/api.go (hunk 4 above)
Comment (talakad) [reply api/api.go]: Thanks. Will fix that

File: api/api.go (hunk 3 above)
Comment (talakad) [reply api/api.go]: Yes. I will add. A volume list that matches both criteria will be used.

File: api/api.go (hunk 2 above)
Comment (talakad) [reply api/api.go]: Yes. Same API is used to delete the schedule as both are recognized by UUID. thought of re-suaing the same API, but input parameters could get confusing. Hence the separate API.

File: api/client/volume/client.go (hunk 1 above)
Comment (talakad) [reply api/client/volume/client.go]: rest of the operations in the same class don' have them defined. so leaving it as it is.

Comment (disrani-px) [reply api/client/volume/client.go]: Actually the new APIs I added do this. We'll never fix this if we keep adding string everywhere

**Pull request merged**